### PR TITLE
Remove leniency during replay translog in peer recovery

### PR DESCRIPTION
**Pull request by dnhatn**
Translog operations replayed during peer recovery should never fail. This change removes leniency in InternalEngine during replaying translog in peer recovery.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/index/replication/ESIndexLevelReplicationTestCase.java [hunk 1]
```diff
@@ -197,7 +197,7 @@ public void backgroundSync(ShardId shardId, RetentionLeases retentionLeases) {
             }
         };
 
-        protected ReplicationGroup(final IndexMetaData indexMetaData) throws IOException {
+        public ReplicationGroup(final IndexMetaData indexMetaData) throws IOException {
```
Comment (ywelsch) [inline test/framework/src/main/java/org/elasticsearch/index/replication/ESIndexLevelReplicationTestCase.java]: is this change necessary? I thought it was accessed from a subclass?

**Review by ywelsch — APPROVED** [review]
LGTM

File: test/framework/src/main/java/org/elasticsearch/index/replication/ESIndexLevelReplicationTestCase.java (hunk 1 above)
Comment (dnhatn) [reply test/framework/src/main/java/org/elasticsearch/index/replication/ESIndexLevelReplicationTestCase.java]: fixed in https://github.com/elastic/elasticsearch/pull/44989/commits/1b91900e403e16acefad70d50ab5009a5a690f3f.

**Comment by dnhatn** [comment]
Thanks @ywelsch.
